build-coordinator/src/main/java/org/jboss/pnc/coordinator/builder/DefaultBuildCoordinator.java

```diff
@@ -187,17 +245,23 @@ private void processBuildTask(BuildTask task) {
                         task.getId(), task.getRebuildAll(), prepareBuildTaskFilterPredicate().test(task), task.getBuildConfigurationAudited().getIdRev());
                 if(!task.getRebuildAll() && prepareBuildTaskFilterPredicate().test(task)) {
                     log.info("[{}] Marking task as REJECTED_ALREADY_BUILT, because it has been already built", task.getId());
-                    task.setStatus(BuildCoordinationStatus.REJECTED_ALREADY_BUILT);
+                    notifyBuildStatusChanged(task, BuildCoordinationStatus.REJECTED_ALREADY_BUILT);
+                    task.setHasFailed(BuildCoordinationStatus.REJECTED_ALREADY_BUILT.hasFailed());
+                    task.setStatus( BuildCoordinationStatus.REJECTED_ALREADY_BUILT);
                     task.setStatusDescription("The configuration has already been built.");
                     markFinished(task);
                     return;
                 }
                 task.setStartTime(new Date());
-                task.setStatus(BuildCoordinationStatus.BUILDING); //status must be updated before startBuild as if build takes 0 time it complete before having Building status.
+                notifyBuildStatusChanged(task,BuildCoordinationStatus.BUILDING);
+                task.setStatus(BuildCoordinationStatus.BUILDING);//status must be updated before startBuild as if build takes 0 time it complete before having Building status.
+                task.setHasFailed(BuildCoordinationStatus.BUILDING.hasFailed());
             }
             buildScheduler.startBuilding(task, onComplete);
         } catch (CoreException | ExecutorException e) {
             log.debug(" Build coordination task failed. Setting it as SYSTEM_ERROR.", e);
+            notifyBuildStatusChanged(task,BuildCoordinationStatus.SYSTEM_ERROR);
```

Comment:
The same as with `notifyBuildSetStatusChanged` - I think it would be better to have all the methods that are executed when the status is changed called from one place.